docker-compose.yml

```diff
@@ -1,91 +1,118 @@
 version: '3.4'
 
-x-kuzzle-config: &kuzzle-config
-  image: kuzzleio/core-dev:2
-  command: sh -c 'chmod 755 /run-dev.sh && /run-dev.sh'
-  volumes:
-    - ".:/var/app"
-    - "./docker/scripts/run-dev.sh:/run-dev.sh"
-    - "./docker/fixtures:/fixtures"
-  cap_add:
-    - SYS_PTRACE
-  ulimits:
-    nofile: 65536
-  sysctls:
-    - net.core.somaxconn=8192
-  depends_on:
-    - redis
-    - elasticsearch
-  environment:
-    - kuzzle_services__storageEngine__client__node=http://elasticsearch:9200
-    - kuzzle_services__storageEngine__commonMapping__dynamic=true
-    - kuzzle_services__internalCache__node__host=redis
-    - kuzzle_services__memoryStorage__node__host=redis
-    - kuzzle_server__protocols__mqtt__enabled=true
-    - kuzzle_server__protocols__mqtt__developmentMode=false
-    - kuzzle_limits__loginsPerSecond=50
-    - NODE_ENV=${NODE_ENV:-development}
-    - DEBUG=${DEBUG:-kuzzle:*,-kuzzle:network:protocols:websocket,-kuzzle:events}
-    # - DEBUG=${DEBUG:-none}
-    - DEBUG_DEPTH=${DEBUG_DEPTH:-0}
-    - DEBUG_MAX_ARRAY_LENGTH=${DEBUG_MAX_ARRAY:-100}
-    - DEBUG_EXPAND=${DEBUG_EXPAND:-off}
-    - DEBUG_SHOW_HIDDEN={$DEBUG_SHOW_HIDDEN:-on}
-    - DEBUG_COLORS=${DEBUG_COLORS:-on}
-    - KUZZLE_VAULT_KEY=${KUZZLE_VAULT_KEY:-secret-password}
-    - KUZZLE_SECRETS_FILE=${KUZZLE_SECRETS_FILE:-/var/app/features/fixtures/secrets.enc.json}
-    # Variables used by the development scripts
-    - NODE_12_VERSION=12.20.0
-    - NODE_14_VERSION=14.17.0
-    - WITHOUT_KUZZLE # Run only Elasticsearch and Redis
-    - REBUILD # Force a rebuild of npm modules
 
 services:
-  nginx:
-    image: nginx:1.19-alpine
-    container_name: kuzzle_nginx
-    depends_on:
-      - kuzzle_node_1
-      - kuzzle_node_2
-      - kuzzle_node_3
+  traefik:
+    image: traefik
+    container_name: traefik
     ports:
       - "7512:7512"
+      - "1883:1883"
+    command:
+      - --log.level=DEBUG
+      - --providers.docker=true
+      - --providers.docker.exposedbydefault=false
+      - --providers.docker.endpoint=unix:///var/run/docker.sock
+      - --entrypoints.kuzzle_http.address=:7512
+      - --entrypoints.kuzzle_mqtt.address=:1883
+    depends_on:
+      kuzzle:
+        condition: service_healthy
     volumes:
-      - ./docker/nginx-dev:/etc/nginx/conf.d
-
-  kuzzle_node_1:
-    <<: *kuzzle-config
-    container_name: kuzzle_node_1
-    ports:
-      - "17510:7512" # Kuzzle API port
-      - "1883:1883" # Kuzzle MQTT port
-      - "9229:9229" # Debug port
+      - "/var/run/docker.sock:/var/run/docker.sock:ro"
 
-  kuzzle_node_2:
-    <<: *kuzzle-config
-    container_name: kuzzle_node_2
-    ports:
-      - "17511:7512" # Kuzzle API port
-      - "11883:1883" # Kuzzle MQTT port
-      - "9230:9229" # Debug port
+  kuzzle_installer:
```

Comment:
I'm not found of this since it will run npm install every time we run Docker-Compose.

This file is meant to be used by core developers anyway so IMHO it's not a problem if it's not ready-to-use but that you have to read the documentation before

Reply:
Yes it runs `npm install` at each `docker-compose up` command BUT since this is not `npm ci` it take 5 seconds to get that no installation is needed. I think this is not a big deal and permit core team non Linux users to run this stack easily

Reply:
In my opinion, we need to separate Core development environments in two types:
- Using this Docker Compose file to install and run all the components
- Using `kourou app:start-services` and local Node.js environment (using Ergol)
As it everyone can contribute easily to the projet

Reply:
But we need to stop hybride workflows that led to packaging issue and false positive during tests

Reply:
I can amend the CONTRIBUTING.md file to explain how to run Kuzzle with each method

Reply:
As a mac user, I welcome this stack that would make it extremely easy to run a kuzzle cluster from scratch with zero-fuss on non-linux systems.
I agree, as Kuzzle developers we are aware of how it works and are of course able to manually perform all the steps manually, but improving the DX is always a good thing, even for core-dev.

Reply:
But, at the same time, I'm not sure it's a good thing that the install performs `npm install`. If dependencies have minor/patch updates, they will be installed and the stack will not be iso-release. This goes without mentioning the risk of potential breaking-in-minor changes...
In the end, I like the idea of having an installation step, but I think we should improve it further.

Reply:
Yes I agree with you, the installation and build process needs some fine tuning